### PR TITLE
Add c_client step in zig build

### PR DESCRIPTION
**Pull request by v0idpwn**
I think this would be useful in general. I'm still not 100% sure about the implementation, and also it needs tests, so I'm opening as a draft to see if it would be welcome.

## Pre-merge checklist

Performance:
* [X] I am very sure this PR could not affect performance.

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -546,3 +553,48 @@ fn dotnet_client(
...
+        "aarch64-macos",
+    };
+
+    inline for (platforms) |platform| {
+        const cross_target = CrossTarget.parse(.{ .arch_os_abi = platform, .cpu_features = "baseline" }) catch unreachable;
+
+        const lib = b.addSharedLibrary("tb_client", "src/clients/c/tb_client.zig", .unversioned);
+        lib.setMainPkgPath("src");
+        // Where should this be output?
+        lib.setOutputDir("zig-out/lib/" ++ platform);
```
Comment (v0idpwn) [inline build.zig]: Plus, this :)

**Comment by batiati** [comment]
Thanks for the PR.
In order to be useful for more use cases, we should output both the dynamically and statically linked libraries, plus `tb_client.h` in some specific dir (e.g. `/src/clients/c/lib`, added to git ignore). 

This output dir could be packed by the CI into a single artifact and made available for download.

**Comment by v0idpwn** [comment]
Updated to include static library + headers and changed output directory. Will look into the CI part later.

File: build.zig [hunk 2]
```diff
@@ -546,3 +553,60 @@ fn dotnet_client(
...
 }
+
+fn c_client(
+    b: *std.build.Builder,
+    mode: Mode,
+    dependencies: []const *std.build.Step,
+    options: *std.build.OptionsStep,
+    tracer_backend: config.TracerBackend,
+) void {
+    const build_step = b.step("c_client", "Build C client shared library");
```
Comment (batiati) [inline build.zig]: Remove the word "shared"

File: build.zig [hunk 3]
```diff
@@ -546,3 +553,60 @@ fn dotnet_client(
...
+    const install_header = b.addInstallFile(
+        .{ .path = "src/clients/c/tb_client.h" },
+        "../src/clients/c/include/tb_client.h",
+    );
+
+    build_step.dependOn(&install_header.step);
+
+    // Zig cross-targets
+    const platforms = .{
+        "x86_64-linux",
```
Comment (batiati) [inline build.zig]: We need to build for both Linux ABI: gnu and musl

File: build.zig [hunk 4]
```diff
@@ -546,3 +553,60 @@ fn dotnet_client(
...
+            lib.setBuildMode(mode);
+            lib.linkLibC();
+
+            if (cross_target.os_tag.? == .windows) {
+                lib.linkSystemLibrary("ws2_32");
+                lib.linkSystemLibrary("advapi32");
+            }
+
+            lib.addOptions("vsr_options", options);
+
```
Comment (batiati) [inline build.zig]: remove this blank line

```suggestion
```

**Review by batiati — CHANGES_REQUESTED** [review]
Great job, thank you for this PR.

It's OK to leave it out of the CI for now. We still have to figure out how we want to publish these artifacts, maybe along with the weekly release, but not sure yet.

File: build.zig [hunk 5]
```diff
@@ -546,3 +553,62 @@ fn dotnet_client(
...
+    const build_step = b.step("c_client", "Build C client library");
+
+    for (dependencies) |dependency| {
+        build_step.dependOn(dependency);
+    }
+
+    // Updates the generated header file:
+    const install_header = b.addInstallFile(
+        .{ .path = "src/clients/c/tb_client.h" },
+        "../src/clients/c/include/tb_client.h",
```
Comment (batiati) [inline build.zig]: Let's keep all artifacts under a single output dir:

```suggestion
        "../src/clients/c/lib/include/tb_client.h",
```

File: .gitignore [hunk 6]
```diff
@@ -1,6 +1,8 @@
 demos/bitcast/bench-decode
 demos/bitcast/decode
 demos/bitcast/transfers
+src/clients/c/lib
+src/clients/c/include
```
Comment (batiati) [inline .gitignore]: No need for this one
```suggestion
```

**Review by batiati — CHANGES_REQUESTED** [review]

**Review by batiati — APPROVED** [review]

**Pull request merged**